ext/date/php_date.c

```diff
@@ -5346,7 +5346,7 @@ static void php_do_date_sunrise_sunset(INTERNAL_FUNCTION_PARAMETERS, bool calc_s
 	if (N > 24 || N < 0) {
 		N -= floor(N / 24) * 24;
 	}
-	if (N > 24 || N < 0) {
+	if (N > 24 || N < 0 || !zend_finite(N)) {
```

Comment:
It's also possible to do it like this: `if (!(N <= 24 && N >= 0)) {`, but yours works too.

Reply:
sure I m gonna try your simpler approach.

Reply:
I think you copied the condition incorrectly

Reply:
yes..